### PR TITLE
Completing conda environments and adding to gitignore

### PR DESCRIPTION
**Pull request by gwaybio**
Here I add the preprocessing conda environment and add some required packages to the segmentation environment.

The additional jupyter-associated packages enable explicit toggling of conda kernel in jupyter lab (not sure what happens in vscode)

I also add some developmental files to gitignore.

### CODE REVIEW

File: 2_segmenting_data/2.segmenting_data.yml [hunk 1]
```diff
@@ -3,6 +3,12 @@ channels:
...
   - conda-forge::pandas=1.4.2
-  - conda-forge::cellpose=2.0.5
\ No newline at end of file
+  - conda-forge::cellpose=2.0.5
+  - conda-forge::scikit-image=0.19.2
+  - conda-forge::matplotlib=3.5.2
+  - conda-forge::jupyter=1.0.0
+  - conda-forge::jupyterlab
+  - conda-forge::ipykernel
+  - conda-forge::nb_conda_kernels
```
Comment (jenna-tomkinson) [inline 2_segmenting_data/2.segmenting_data.yml:13]: Just a question, what is this dependency for? I haven't seen it used before.

File: 2_segmenting_data/2.segmenting_data.yml [hunk 2]
```diff
@@ -3,6 +3,12 @@ channels:
...
+  - conda-forge::jupyter_contrib_nbextensions
```
Comment (jenna-tomkinson) [inline 2_segmenting_data/2.segmenting_data.yml:14]: Same with this one.

**Review by jenna-tomkinson — APPROVED** [review]
I love the addition of a conda environment for `PyBaSiC`! I am excited to try it on my end. 

I only have a few questions in regard to two dependencies, but it all LGTM!

File: 2_segmenting_data/2.segmenting_data.yml (hunk 1 above)
Comment (gwaybio) [reply 2_segmenting_data/2.segmenting_data.yml:13]: This is an important package for using `jupyter lab` in terminal. It enables me to explicitly set the conda kernel

https://github.com/Anaconda-Platform/nb_conda_kernels

> This extension enables a [Jupyter Notebook](http://jupyter.org/) or [JupyterLab](https://jupyterlab.readthedocs.io/en/stable/) application in one [conda](https://conda.io/docs/) environment to access kernels for Python, R, and other languages found in other environments. When a kernel from an external environment is selected, the kernel conda environment is automatically activated before the kernel is launched. This allows you to utilize different versions of Python, R, and other languages from a single Jupyter installation.

File: 2_segmenting_data/2.segmenting_data.yml (hunk 2 above)
Comment (gwaybio) [reply 2_segmenting_data/2.segmenting_data.yml:14]: This package adds some functionality to jupyter UI: https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/

I'm not actually sure if I ever use any of the extensions explicitly though!

**Comment by gwaybio** [comment]
woo!

This comment got me thinking, why not add pybasic to the conda environment file itself!

> I love the addition of a conda environment for PyBaSiC! I am excited to try it on my end.

I tried by adding this to the end of the .yml file:

```yaml
- pip:
    - git+https://github.com/peng-lab/BaSiCPy@f3fcf1987db47c4a29506d240d0f69f117c82d2b
```

But received the following error. We'll need to keep our current install instructions in the README and update once the PyBaSiC folks get a chance to address (we can also contribute directly, but I suspect this is a longer route).

```bash
Pip subprocess error:
  Running command git clone --filter=blob:none --quiet https://github.com/peng-lab/BaSiCPy /private/var/folders/02/q30k_4wn2dqbz5pj_vvc8xn40000gp/T/pip-req-build-m7r31ud2
  Running command git rev-parse -q --verify 'sha^f3fcf1987db47c4a29506d240d0f69f117c82d2b'
  Running command git fetch -q https://github.com/peng-lab/BaSiCPy f3fcf1987db47c4a29506d240d0f69f117c82d2b
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      error: Multiple top-level packages discovered in a flat-layout: ['figures', 'pybasic', 'Notebooks', 'ExampleData'].

      To avoid accidental inclusion of unwanted files or directories,
      setuptools will not proceed with this build.

      If you are trying to create a single distribution with multiple packages
      on purpose, you should not rely on automatic discovery.
      Instead, consider the following options:

      1. set up custom discovery (`find` directive with `include` or `exclude`)
      2. use a `src-layout`
      3. explicitly set `py_modules` or `packages` with a list of names

      To find more information, look for "package discovery" on setuptools docs.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

failed

CondaEnvException: Pip failed
```

I'm going to merge!

**Pull request merged**